hw3/react-hw3/thrones/src/Components/Search/SearchComponent.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+
+import InputGroup from "react-bootstrap/InputGroup";
+import axios from "axios";
+import { Card, Row, Col } from 'react-bootstrap';
+
+export class SearchComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: "",
+      character: null,
+      notFound: false,
+      multiple: false,
+      searchItem: {},
+      searchItemFound: false,
+    };
+  }
+
+  filterData = (event) => {
+    const foundData = this.state.character.filter((item) =>
+      item.fullName.toLowerCase().includes(this.state.query.toLowerCase())
+    );
+    if (foundData.length !== 0) {
+      this.setState({
+        searchItemFound: true,
+        searchItem: foundData,
+        notFound: false,
+        multiple: false,
+      });
+    } else {
+      this.setState({
+        searchItemFound: false,
+        searchItem: {},
+        notFound: true,
+        multiple: false,
+      });
+    }
+  };
+
+  handleEnterButtonClick = (event) => {
+    if (event.keyCode === 13) {
+      this.filterData(event);
+    } else {
+    }
+  };
+
+  handleInputChange = (event) => {
+    this.setState({ query: event.target.value });
+  };
+
+  componentDidMount() {
+    this.searchCharacter();
+  }
+
+  searchCharacter = async () => {
+    try {
+      const response = await axios.get(
+        `https://thronesapi.com/api/v2/Characters`
+      );
+      if (response.data.length > 1) {
+        this.setState({
+          character: response.data,
+          notFound: false,
+          multiple: false,
+        });
+      } else {
+        this.setState({ character: null, notFound: true, multiple: false });
+      }
+    } catch (error) {
+      console.error("Error fetching character data:", error);
+    }
+  };
```

Comment:
Calls the same API and fetches the same data as that of HousesComponent.js

Fetching the data (calling the exact same API) twice, in both the Houses and Search components is inefficient. The data they both rely on is the same, and additionally both of these components have the same parent.

So why not let the data flow from parent to child?

It would make a lot more sense to call the API once in the App component, then pass the data as props to the houses and search components. This would be more efficient since the API would only need to be called once, but the data would be available across the components that need it.

In summary, you can just fetch the data once and then pass it to child components via props.